File: Demo Projects/volley_library_demo/app/src/main/java/com/example/volley_library_demo/MainActivity.java
package com.example.volley_library_demo;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.util.Log;

import com.android.volley.Request;
import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.JsonObjectRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        RequestQueue requestQueue;//from volley library    it makes a queue of request
        requestQueue = Volley.newRequestQueue(this);

        //json object request is used to add the request into the list
        JsonObjectRequest jsonObjectRequest=new JsonObjectRequest(Request.Method.GET, "'https://randomuser.me/api/'",
                null, new Response.Listener<JSONObject>() {
            @Override
            public void onResponse(JSONObject response) {
                try {
                    Log.d("myapp","The response is "+response.getString("title")); //from the output of the api we know this
                } catch (JSONException e) {
                    e.printStackTrace();
                }
            }
        }, new Response.ErrorListener() {
            @Override
            public void onErrorResponse(VolleyError error) {
                Log.d("myapp","Something went wrong");
            }
        });

        requestQueue.add(jsonObjectRequest);
    }
}

File: Latitude_projects/RoomDemoFinal/app/src/main/java/com/example/roomdemofinal/MainActivity.kt
package com.example.roomdemofinal

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import androidx.lifecycle.ViewModelProvider
import com.example.roomdemofinal.api.ApiInterface
import com.example.roomdemofinal.api.ApiUtilities
import com.example.roomdemofinal.repository.PostsRepository
import com.example.roomdemofinal.viewModel.MemesViewModelFactory
import com.example.roomdemofinal.viewModel.PostsViewModel

class MainActivity : AppCompatActivity() {
    private  lateinit var  postsViewModel: PostsViewModel
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)



        val apiInterface = ApiUtilities.getInstance().create(ApiInterface::class.java)

        val postRepository = PostsRepository(apiInterface)

        postsViewModel= ViewModelProvider(this,MemesViewModelFactory(postRepository)).get(PostsViewModel::class.java)

        postsViewModel.posts.observe(this,{
            Log.d("dolorem","onCreatejk:${it.toString()}")
        })


    }
}